tox.ini

```diff
@@ -13,55 +13,40 @@ envlist =
 requires =
     setuptools >= 30.3.0
     pip >= 19.3.1
-    tox-pypi-filter >= 0.12
-
-[testenv]
-pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/main/pip_pinnings.txt
 
 extras =
     test
-    all
-
-commands =
-    pip freeze
-    pytest --pyargs pyvo
+    alldeps: all
 
-# tox environments are constructed with so-called 'factors' (or terms)
-# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
-# will only take effect if that factor is included in the environment name. To
-# see a list of example environments that can be run, along with a description,
-# run:
-#
-#     tox -l -v
-#
 description =
     run tests
-    oldestdeps: with astropy 4.1.*
-    devastropy: with astropy latest
+    oldestdeps: with oldest supported dependencies
+    devdeps: with development version of dependencies
+    cov: determine the code coverage
 
 deps =
-    devastropy: git+https://github.com/astropy/astropy.git#egg=astropy
+    cov: coverage
+
     oldestdeps: astropy==4.1
     # We set a suitably old numpy along with an old astropy, no need to pick up
     # deprecations and errors due to their unmatching versions
     oldestdeps: numpy==1.16
 
-[testenv:egg_info]
-description = ensure egg_info works without dependencies
 commands =
+    devdeps: pip install -U --pre --only-binary :all: -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
```

Comment:
Stuart proposed another way to do this over at https://github.com/astropy/astropy/pull/13614

Reply:
Hmm, interesting. That was more like the first approach we had, and then we switched over to having these individual pip installs. I think Stuart's version is cleaner in the case when you have multiple packages. However, here we only use the scipy-wheels-nightly for numpy, and something else for astropy, and don't install other packages. So I would say we should keep it as a one-liner like this rather than adding extras.

Reply:
There's very little to choose between them, so use your favourite, but sunpy uses multiple index URLs with the same pattern https://github.com/sunpy/sunpy/blob/main/tox.ini#L34